packages/web-pkg/src/composables/appDefaults/useAppDocumentTitle.ts

```diff
@@ -5,27 +5,31 @@ import { useAppMeta } from './useAppMeta'
 import { useDocumentTitle } from './useDocumentTitle'
 import { Store } from 'vuex'
 import { MaybeRef } from '../../utils'
+import { Route } from 'vue-router'
 
 interface AppDocumentTitleOptions {
   store: Store<any>
   applicationId: string
   applicationName?: MaybeRef<string>
   currentFileContext: Ref<FileContext>
+  currentRoute?: Ref<Route>
 }
 
 export function useAppDocumentTitle({
   store,
   applicationId,
   applicationName,
-  currentFileContext
+  currentFileContext,
+  currentRoute
 }: AppDocumentTitleOptions): void {
   const appMeta = useAppMeta({ applicationId, store })
 
   const titleSegments = computed(() => {
-    const fileName = basename(unref(unref(currentFileContext).fileName))
+    const baseTitle =
+      basename(unref(unref(currentFileContext).fileName)) || unref(currentRoute)?.meta?.title
```

Comment:
```suggestion
      basename(unref(unref(currentFileContext).fileName)) || $gettext(unref(currentRoute)?.meta?.title)
```

The route titles are statically defined and only ever wrapped in a dummy `$gettext` helper that ensures string **extraction** to transifex. The dummy helpers never do actual runtime translations, thus you to feed the respective string into the actual `$gettext` function.